### DISCUSSION TITLE
How to run it on my own linux server?

### DISCUSSION THREAD

**Discussion by JenuelDev** [open-webui/open-webui#2608] [Q&A] [answered]
how to run it on my own linux server, and access it anywhere I want?

**Comment by JuanjoJG-WATA** [comment, marked as answer]
You need to follow the steps on the documentation. Basically, the process goes like this:

-Clone the repo
-Copy the .env file
-Build the frontend
-Serve the frontend with the backend
-Download ollama
-Download a model
-Run the start.sh
-Done

Here you have the full documentation: https://github.com/open-webui/open-webui/tree/functions?tab=readme-ov-file#how-to-install-without-docker

(Mind you, this way is WITHOUT Docker)

**Reply by JuanjoJG-WATA** [reply]
On top of that, if you want to have it always running, you can create a service with **systemd**, so everytime you start your server, the WebUI will start too.

Also, don't forget you need to have Bun or Node.js aswell as Python, and they all need to be updated.

**Reply by hectibrown** [reply]
> On top of that, if you want to have it always running, you can create a service with **systemd**, so everytime you start your server, the WebUI will start too.
> 
> Also, don't forget you need to have Bun or Node.js aswell as Python, and they all need to be updated.

Hey, I have open-webui in an LXC and it runs great, but for my .service file, what needs to go in there? I asked my ai and it provided this template, but i am not sure where my executable for "serve" is...

```
[Unit]

Description=Open-WebUI Serve


[Service]

Type=simple

ExecStart=/path/to/your/open-webui-serve-executable

Restart=always

User=your-username


[Install]

WantedBy=multi-user.target
```

**Reply by JuanjoJG-WATA** [reply]
Sorry about the delay in answering, I'm on vacation lol. Here's the one I used:

[Unit]
Description=WebUI Service
After=network.target
[Service]
Type=simple
ExecStart=/root/open-webui/backend/start.sh
Restart=always
RestartSec=3
StandardOutput=syslog
StandardError=syslog
[Install]
WantedBy=multi-user.target

**Reply by hectibrown** [reply]
> Sorry about the delay in answering, I'm on vacation lol. Here's the one I used:
> 
> [Unit] Description=WebUI Service After=network.target [Service] Type=simple ExecStart=/root/open-webui/backend/start.sh Restart=always RestartSec=3 StandardOutput=syslog StandardError=syslog [Install] WantedBy=multi-user.target

Okay, so i had no luck, and since i installed open-webui with pipx i can't find the start.sh file anywhere.... I installed via PIPX because PIP was giving me issue... 

```
root@OpenWebUi:~# find -name open-webui
./.local/pipx/venvs/open-webui
./.local/pipx/venvs/open-webui/bin/open-webui
./.local/bin/open-webui
```

when I do a cd into any of the open-webui above it tell me it is not a directory...

edit:
they are python files... not sure what to do now

**Comment by ami5451** [comment]
My systemctl service config for open-webui is as follows.

1. I compiled my own Python3 (3.11.10) and Nodejs (v20.9.0) from source because I wanted to use the latest supported versions (and the versions that came with the O/S were a little older), which is why I updated the PATH in my .bash_profile to ensure my versions of Python3 and Nodejs are encountered first before any default system versions. 
2. I installed the latest version of Anaconda/conda available for my Linux system ( https://repo.anaconda.com/archive/ )
3. During the creation of the conda environment, I chose the option for conda to automatically initialize, so conda inserted relevant entries into my /home/supportu/.bashrc file
4. I used the manual installation method from https://docs.openwebui.com/getting-started/
5. I used mostly default settings e.g., my conda environment is called open-webui-env
6. I created the systemd service file for open-webui as /etc/systemd/system/open-webui.service
7. I pointed the service ExecStart to a custom file rather than directly to the open-webui start.sh (this allowed me to initialize the environment properly before calling the open-webui start.sh)
8. Enabled the open-webui service so that it runs at boot (systemctl enable open-webui)
9. I am able to stop and start the open-webui service using systemctl

Below are the contents of the various files (adapt to your your setup as necessary):

$ cat /home/supportu/.bash_profile
if [ -f ~/.bashrc ]; then
        . ~/.bashrc
fi
PATH=/opt/python/3.11.10/bin:/usr/local/bin:$PATH
export PATH


$ cat /home/supportu/systemd-start-openwebui.sh
#!/bin/bash
source /home/supportu/.bash_profile
source /home/supportu/.bashrc
conda activate open-webui-env
/home/supportu/open-webui/backend/start.sh


$ sudo cat /etc/systemd/system/open-webui.service
[Unit]
Description=WebUI Service
After=network.target

[Service]
Type=simple
ExecStart=/home/supportu/systemd-start-openwebui.sh
KillSignal=SIGTERM
KillMode=mixed
Restart=always
RestartSec=3
StandardOutput=syslog
StandardError=syslog
User=supportu

[Install]
WantedBy=multi-user.target``

**Comment by IncubuzzCC** [comment]
Hi,
I am aware that this post is old and unfortunately it is the main one showing in my searches.
The way I got it to run as a non root user is actually much more simple:
Edit your service file as root /etc/systemd/system/open-webui.service
```
[Unit]
Description=Open WebUI Service
After=network.target

[Service]
Type=simple
WorkingDirectory=/home/webui
ExecStart=/home/webui/.local/bin/open-webui serve
ExecStop=/bin/kill -HUP $MAINPID
User=webui
Group=webui

[Install]
WantedBy=multi-user.target
```

Then reload and start your systenctrl daemon:
```
sudo systemctl daemon-reload
sudo systemctl enable open-webui
sudo systemctl start open-webui
sudo systemctl status open-webui
```

**Comment by Tronic** [comment]
I found it much easier to run with [UV](https://docs.astral.sh/uv/getting-started/installation/), only adding `/etc/systemd/system/open-webui.service`:

```systemd
[Unit]
Description=Open WebUI Service
After=network.target

[Service]
Type=simple
ExecStart=uvx open-webui serve --port 4100
User=openwebui
Restart=always
RestartSec=5
WorkingDirectory=/srv/open-webui

[Install]
WantedBy=multi-user.target
```

You'll need to install UV on the system and `useradd -rmd /srv/open-webui openwebui` but don't need to install anything else. For self hosted models possibly want to install ollama too but that comes prepackaged and doesn't need updating every week like open-webui (don't worry, uvx will handle that for you every time the service is restarted).

To make it publicly available, you'll want to install [Caddy](https://caddyserver.com/) and in `/etc/caddy/Caddyfile`:

```Caddyfile
yourdomain.example.com {
    reverse_proxy :4100
}
```

This will automatically obtain TLS certificate on that domain and connect to localhost:4100 openwebui backend by plain HTTP. In case you cannot run that at home (e.g. your ISP doesn't give you a public IP), you can also run Caddy at a virtual private server, and setup WireGuard to have it connect to your home machine through any firewalls (but that is outside the scope of my post).
